packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -543,41 +580,61 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 		}
 
 		// Deal with gw wide zlib-stream compression
-		if (this.inflate) {
-			const l = decompressable.length;
+		if (this.strategy.options.compression !== null) {
 			const flush =
-				l >= 4 &&
-				decompressable[l - 4] === 0x00 &&
-				decompressable[l - 3] === 0x00 &&
-				decompressable[l - 2] === 0xff &&
-				decompressable[l - 1] === 0xff;
+				decompressable.length >= 4 &&
+				decompressable.at(-4) === 0x00 &&
+				decompressable.at(-3) === 0x00 &&
+				decompressable.at(-2) === 0xff &&
+				decompressable.at(-1) === 0xff;
 
-			const zlib = (await getZlibSync())!;
-			this.inflate.push(Buffer.from(decompressable), flush ? zlib.Z_SYNC_FLUSH : zlib.Z_NO_FLUSH);
+			if (this.nativeInflate) {
+				this.nativeInflate!.write(decompressable, 'binary');
```

Comment:
And if that `!` is needed, I will yell at TS

```suggestion
				this.nativeInflate.write(decompressable, 'binary');
```